storage-node/src/commands/server.ts

```diff
@@ -219,16 +228,25 @@ Supported values: warn, error, debug, info. Default:debug`,
     const enableUploadingAuth = false
     const operatorRoleKey = undefined
 
-    await recreateTempDirectory(flags.uploads, TempDirName)
-
-    if (fs.existsSync(flags.uploads)) {
-      await loadDataObjectIdCache(flags.uploads, TempDirName, PendingDirName)
+    if (!flags.tempFolder) {
+      logger.warn(
+        'It is recommended to specify a unique file path for temporary files.' +
+          'For now a temp folder under the uploads folder will be used.' +
+          'In future this will be warning will become and error!'
+      )
     }
 
-    if (flags.dev) {
-      await this.ensureDevelopmentChain()
+    const tempFolder = flags.tempFolder || path.join(flags.uploads, TempDirName)
+
+    if (path.relative(tempFolder, flags.uploads) === '') {
+      this.error('Paths for temporary and uploads folders must be unique.')
```

Comment:
```suggestion
    if (path.relative(tempFolder, flags.uploads) === '..') {
      this.error('Paths for temporary and uploads folders must be unique.')
```

Please see the other comment

Reply:
commented here https://github.com/Joystream/joystream/pull/5000#discussion_r1435673895